internal/pd/get-shifts.go

```diff
@@ -0,0 +1,115 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package pd
+
+import (
+	"strconv"
+)
+
+//Requirement: Shifts must be sorted in .pd.yml file
+
+//Shift specifies time range and name of a shift
+type Shift struct {
+	Start Time
+	End   Time
+	Name  string
+}
+
+//Time stores hour and minute of a specific moment
+type Time struct {
+	Hour   int
+	Minute int
+}
+
+//GetNextShift returns information about the next shift and time until it starts
+func GetNextShift(shifts []Shift, shiftPos int, currentTime Time) (Shift, Time, error) {
+
+	var nextShift Shift
+	if shiftPos+1 < len(shifts) {
+		nextShift = shifts[shiftPos+1]
+	} else {
+		nextShift = shifts[0]
+	}
+
+	timeUntilNextShift := Time{0, 0}
+
+	timeUntilNextShift.Hour = nextShift.Start.Hour - currentTime.Hour
+	if currentTime.Minute > nextShift.Start.Minute {
+		timeUntilNextShift.Hour--
+		timeUntilNextShift.Minute = 60 - currentTime.Minute + nextShift.Start.Minute
+	} else {
+		timeUntilNextShift.Minute = nextShift.Start.Minute - currentTime.Minute
+	}
+	if currentTime.Hour > nextShift.Start.Hour {
+		timeUntilNextShift.Hour += 24
+	}
+
+	return nextShift, timeUntilNextShift, nil
+}
+
+//GetCurrentShift returns all shifts in a slice and the position of the current shift
+func GetCurrentShift(currentTime Time) ([]Shift, int, error) {
+
+	shifts, err := LoadShifts()
+	if err != nil {
+		return []Shift{}, 0, err
+	}
+
+	var shiftPos int
+	for i, shift := range shifts {
+		if shift.Start.Hour < shift.End.Hour {
+			if currentTime.Hour > shift.Start.Hour && currentTime.Hour < shift.End.Hour || currentTime.Hour == shift.Start.Hour && currentTime.Minute >= shift.Start.Minute || currentTime.Hour == shift.End.Hour && currentTime.Minute < shift.End.Minute {
+				shiftPos = i
+			}
+		} else if shift.Start.Hour > shift.End.Hour {
+			if currentTime.Hour > shift.Start.Hour || currentTime.Hour < shift.End.Hour || currentTime.Hour == shift.Start.Hour && currentTime.Minute >= shift.Start.Minute || currentTime.Hour == shift.End.Hour && currentTime.Minute < shift.End.Minute {
+				shiftPos = i
+			}
+		} else if currentTime.Hour == shift.Start.Hour {
+			if currentTime.Minute >= shift.Start.Minute && currentTime.Minute < shift.End.Minute {
+				shiftPos = i
+			}
+		}
+	}
```

Comment:
Like discussed, this is absolutely fine and does what it has to do. It could be hard to read in one year from now. Try to come up with a bit of a refactoring that this is a bit faster to read. Also, add a couple of comments at the main if cases to provide a bit context.